src/actions/accountsActions.js

```diff
@@ -269,23 +275,31 @@ export const switchAccountAction = (accountId: string) => {
   };
 };
 
-export const initOnLoginSmartWalletAccountAction = (privateKey: string) => {
+export const initOnLoginSmartWalletAccountAction = (
+  privateKey: string,
+  setAccountActive: boolean = true,
+) => {
   return async (dispatch: Dispatch, getState: GetState) => {
     const {
       appSettings: { data: { blockchainNetwork } },
       accounts: { data: accounts },
     } = getState();
 
-    const activeAccountId = getActiveAccountId(accounts);
-    const activeAccountType = getActiveAccountType(accounts);
-    await dispatch(initSmartWalletSdkAction(privateKey));
+    const inactiveAccounts = getInactiveUserAccounts(accounts);
+    const activeAccount = getActiveAccount(accounts);
+    const smartWalletAccount = activeAccount && getAccountType(activeAccount) === ACCOUNT_TYPES.SMART_WALLET
```

Comment:
What if we simply leave:
```suggestion
    const smartWalletAccount = findFirstSmartAccount(accounts);
```
?

Reply:
Makes sense 👍